src/mono/wasm/features.md

```diff
@@ -2,36 +2,311 @@
+
+```
+[JsonSerializable(typeof(List<Item>))]
+partial class ItemListSerializerContext : JsonSerializerContext { }
+
+var json = JsonSerializer.Serialize(items, ItemListSerializerContext.Default.ListItem);
+
+```
+
+WARNING: Make sure that you tested trimmed/published release version of your application.
+
+See also [trimming guidance](https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trim-self-contained)
+
+## C code or native linked libraries
+
+Native rebuild will cause the .NET runtime to be re-built alongside your application, which allows you to link additional libraries into the WASM binary or change compiler configuration flags.
+
+You can enable native rebuild by `<WasmBuildNative>true</WasmBuildNative>`.
+
+You can add C files into the compilation by `<NativeFileReference Include="fibonacci.c" />`.
+
+This requires that you have [wasm-tools workload](#wasm-tools-workload) installed.
+
+# JavaScript API, Interop
+
+We maintain description of JavaScript embedding API in [dotnet.d.ts](https://github.com/dotnet/runtime/blob/main/src/mono/wasm/runtime/dotnet.d.ts).
+
+## Browser application template
+
+You can create simple application template by running `dotnet new wasmbrowser`.
+
+Then you could `dotnet run` and open the URL which it printed to test the app in your browser. For example `http://localhost:5292/index.html`
+
+You can also `dotnet publish -c Release` which will publish your app to [AppBundle](#Project-folder-structure) folder.
+
+## JavaScript interoperability
+
+When you want to call JavaScript functions from C# or managed code from JavaScript, you can annotate static C# methods with `[JSImport]` or `[JSExport]` attributes.
+
+You can read [blog article](https://devblogs.microsoft.com/dotnet/use-net-7-from-any-javascript-app-in-net-7/) about it.
+
+You can also explore simple [TODO app sample](https://github.com/pavelsavara/dotnet-wasm-todo-mvc).
+
+For more details please see [documentation](https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/import-export-interop).
+
+## Embedding dotnet in existing JavaScript applications
+
+The npm tool chains usual in the JavaScript community are not well integrated with the MSBuild tool chain of the dotnet.
+Your project will need to handle both build systems.
+
+You will need pass output of the MSBuild from the `AppBundle` folder for the npm toolchain to process and host it on your web server with the other assets of your application.
+
+You can also explore simple sample of dotnet as [React component](https://github.com/maraf/dotnet-wasm-react).
+
+# Downloaded assets
+
+## Project folder structure
+
+Described as relative to simple application project.
+
+- `./wwwroot` is optional project folder, into which you can place files which should be also deployed to the web server.
+- `./bin/Release/net8.0/browser-wasm/AppBundle` folder which should be hosted by the HTTP server.
```

Comment:
```suggestion
### Build output directories

The following shows the structure for a Release build, but except the name in the various paths, the rest is applicable for a `Debug` build too.

- `./bin/Release/net8.0/browser-wasm/AppBundle` folder which should be hosted by the HTTP server.
```

Reply:
I see, splitting it with headers makes sense